e2e-tests/src/main.rs

```diff
@@ -248,3 +276,68 @@ fn accounts(seeds: Option<Vec<String>>) -> Vec<sr25519::Pair> {
     });
     seeds.into_iter().map(keypair_from_string).collect()
 }
+
+fn get_first_two_accounts(accounts: &[sr25519::Pair]) -> (sr25519::Pair, sr25519::Pair) {
+    let first = accounts.get(0).expect("No accounts passed").to_owned();
+    let second = accounts
+        .get(1)
+        .expect("Pass at least two accounts")
+        .to_owned();
+    (first, second)
+}
+
+#[derive(Debug)]
+struct FeeInfo {
+    base_fee: Balance,
+    len_fee: Balance,
+    unadjusted_weight: Balance,
+    adjusted_weight: Balance,
+    tip: Balance,
+}
+
+fn get_tx_fee_info(connection: &Connection, tx: &TransferTransaction) -> FeeInfo {
+    let block = connection.get_block::<Block>(None).unwrap().unwrap();
+    let block_hash = block.header.hash();
+
+    let unadjusted_weight = connection
+        .get_payment_info(&tx.hex_encode(), Some(block_hash))
+        .unwrap()
+        .unwrap()
+        .weight as Balance;
+
+    let fee = connection
+        .get_fee_details(&tx.hex_encode(), Some(block_hash))
+        .unwrap()
+        .unwrap();
+    let inclusion_fee = fee.inclusion_fee.unwrap();
+
+    FeeInfo {
+        base_fee: inclusion_fee.base_fee,
+        len_fee: inclusion_fee.len_fee,
+        unadjusted_weight,
+        adjusted_weight: inclusion_fee.adjusted_weight_fee,
+        tip: fee.tip,
+    }
+}
+
+fn balance_of(account: &AccountId32, connection: &Connection) -> Balance {
+    connection.get_account_data(account).unwrap().unwrap().free
+}
+
+fn transfer(target: &AccountId32, value: u128, connection: &Connection) -> TransferTransaction {
```

Comment:
apply this function also to lines 108 to 119

Reply:
good catch, done